app/gui/src/presenter/project.rs

```diff
@@ -336,6 +425,22 @@ impl Project {
         self
     }
 
+    fn attach_frp_to_project_list_notifications(
+        self,
+        notifications: Publisher<ProjectListNotification>,
+        project_list_ready: frp::Source<()>,
+    ) -> Self {
+        let weak = Rc::downgrade(&self.model);
+        let notifications = notifications.subscribe();
+        spawn_stream_handler(weak, notifications, move |notification, _| {
+            match notification {
+                ProjectListNotification::ProjectListReady => project_list_ready.emit(()),
+            }
+            std::future::ready(())
+        });
+        self
+    }
```

Comment:
I feel it's a bit strange we make a publisher to publish notifications for ourselves. Cannot we just call "reset_entries" on project list in the async block where we wait for new list load?

Reply:
Yep, I guess it's a lot easier :) Done.